app/scripts/views/mixins/last-checked-time-mixin.js

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Mixin to set the time a panel was refreshed or checked.
+ *
+ * @mixin LastCheckedTimeMixin
+ */
+'use strict';
+
+const {t} = require('../base');
+
+const Mixin = {
+
+  setInitialContext(context) {
+    context.set({
+      lastCheckedTime: this.getLastCheckedTimeString()
+    });
+  },
+
+  setLastCheckedTime(date) {
+    if (! date) {
+      date = new Date();
+    }
+    this.lastCheckedTime = date.toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit'
+    });
+  },
+
+  getLastCheckedTimeString() {
+    let time = t('none');
+    if (this.lastCheckedTime) {
+      time = this.lastCheckedTime;
+    }
+    return t('Last checked') + `: ${time}`;
```

Comment:
Localizers get a bit twitchy when we do string concatenation to form strings, sometimes the format is different for their locale. I'd suggest doing something like:

```js
return this.translate(t('Last checked: %(lastCheckTime)s'), { lastCheckTime: time );
```

This way the localizers have control over how to best form that string for their particular locale.